source/common/runtime/runtime_impl.cc

```diff
@@ -121,6 +121,19 @@ bool SnapshotImpl::featureEnabled(absl::string_view key,
     percent = default_value;
   }
 
+#ifndef NDEBUG
+  // When numerator > denominator condition is always evaluates to TRUE
+  // It becomes hard to debug why configuration does not work in case of wrong numerator.
+  // Log debug message that numerator is invalid.
+  uint64_t denominator_value =
+      ProtobufPercentHelper::fractionalPercentDenominatorToInt(percent.denominator());
+  if (percent.numerator() > denominator_value) {
+    throw EnvoyException(fmt::format("WARNING runtime key '{}': numerator ({}) > denominator ({}), "
```

Comment:
I'd remove the WARNING part of this. You're halting the show with that exception, so it's not really a warning :)

Reply:
Oh yeah, sure, makes sense (I've just copy pasted it)